tests/io_test.py

```diff
@@ -238,13 +235,80 @@ def testGetDataSets(self):
         :raise: AssertionError if the input and returned dataset names don't match
         '''
         self._create_file(columns=self.dict_columns, 
-                          path_prefix='{}/iotest_dict_columns'.format(IOTest.io_test_dir))     
+                          prefix_path='{}/iotest_dict_columns'.format(IOTest.io_test_dir))     
         datasets = ak.get_datasets('{}/iotest_dict_columns_LOCALE0'.format(IOTest.io_test_dir))
 
         self.assertEqual(3, len(datasets)) 
         for dataset in datasets:
             self.assertIn(dataset, self.names)
 
+    def testSaveStringsDataset(self):
+        # Create, save, and load Strings dataset
+        strings_array = ak.array(['string {}'.format(num) for num in list(range(1,11))])
+        strings_array.save('{}/strings-test'.format(IOTest.io_test_dir), dataset='strings')
+        r_strings_array = ak.load_all('{}/strings-test'.format(IOTest.io_test_dir))['strings']
+
+        self.assertEqual(len(strings_array), len(r_strings_array))
+        for i in list(range(0,10)):
+            self.assertEqual(strings_array[i], r_strings_array[i])
+
+        # Read a part of a saved Strings dataset from one hdf5 file
+        r_strings_subset = ak.read_all(filenames='{}/strings-test_LOCALE0'.\
+                                    format(IOTest.io_test_dir))
+        self.assertIsNotNone(r_strings_subset)
+        self.assertTrue(isinstance(r_strings_subset[0], str))    
+        self.assertIsNotNone(ak.read_hdf(filenames='{}/strings-test_LOCALE0'.\
+                            format(IOTest.io_test_dir), dsetName='strings/values'))
+        self.assertIsNotNone(ak.read_hdf(filenames='{}/strings-test_LOCALE0'.\
+                            format(IOTest.io_test_dir), dsetName='strings/segments'))
+
+    def testSaveMixedStringsDataset(self):
+        strings_array = ak.array(['string {}'.format(num) for num in list(range(1,11))])
+        m_floats =  ak.array([x / 10.0 for x in range(0, 10)])      
+        m_ints = ak.array(list(range(0, 10)))
+        ak.save_all({'m_strings': strings_array,
+                     'm_floats' : m_floats,
+                     'm_ints' : m_ints}, 
+                     '{}/multi-type-test'.format(IOTest.io_test_dir))
+        r_mixed = ak.load_all('{}/multi-type-test'.format(IOTest.io_test_dir))
+        for i in list(range(0,10)):
+            self.assertEqual(strings_array[i], r_mixed['m_strings'][i])
+        self.assertEqual(m_floats.all(), r_mixed['m_floats'].all())
+        self.assertEqual(m_ints.all(), r_mixed['m_ints'].all())
```

Comment:
These don't perform the correct checks. They should be:
```python
self.assertTrue((m_floats == r_mixed['m_floats']).all())
self.assertTrue((m_ints == r_mixed['m_ints']).all())
```

As written, the test will pass even if the arrays have different values, as long as they both contain zero (or both do not contain zero).